src/mesh/Router.cpp

```diff
@@ -424,6 +424,9 @@ meshtastic_Routing_Error perhapsEncode(meshtastic_MeshPacket *p)
 
     // If the packet is not yet encrypted, do so now
     if (p->which_payload_variant == meshtastic_MeshPacket_decoded_tag) {
+        p->decoded.has_ok_to_mqtt = true;
+        p->decoded.ok_to_mqtt = config.lora.config_ok_to_mqtt;
```

Comment:
Here, based on your setting, you decide whether other people's packets will be OK to MQTT.

Reply:
As many of these as I added, it's not surprising I got the logic wrong in one of them.

Reply:
> Here, based on your setting, you decide whether other people's packets will be OK to MQTT.

I was under the impression that a packet that got to this point, still decoded, would have to come from a local source, either phone or firmware. How does this code path actually get taken?

Reply:
We try to decrypt every received packet (in `Router::handleReceived()`), because we might need to do something with the payload. Then the only place it gets encrypted again is here, just before transmitting. So every packet we could decrypt would still be decrypted here.

Reply:
Got it. So just need to add the check that it is from the local node.

Reply:
Yes. And if you have it here, it doesn't need to be set also in the modules.